File: LeetCode_Algorithm/1480/1480.py
from typing import List

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:

        # sol1 O(n)
        # 54ms 65%
        # ans = []
        # temp = 0
        # for num in nums:
        #     temp += num
        #     ans.append(temp)
        # return ans


        # sol2 O(n)
        # 33ms 99%
        for idx in range(1, len(nums)):
            nums[idx] = nums[idx-1] + nums[idx]
        return nums

print(Solution().runningSum([1,2,3,4]))

File: LeetCode_Algorithm/1480/1480.js
var runningSum = function(nums) {
  // sol1 O(n)
  // 68ms 86%
  // let ans = []
  // let temp = 0
  // for (let num of nums) {
  //   temp += num
  //   ans.push(temp)
  // }
  // return ans

  // sol2 O(n)
  // 123ms 6%
  for (let idx=1; idx < nums.length; idx++) {
    nums[idx] = nums[idx-1]+nums[idx]
  }
  return nums
};

console.log(runningSum([1,2,3,4]))